kafka/record/util.py

```diff
@@ -0,0 +1,112 @@
+import six
+import binascii
+
+from ._crc32c import crc as crc32c_py
+
+
+if six.PY3:
+    def _read_byte(memview, pos):
+        """ Read a byte from memoryview as an integer
+
+            Raises:
+                IndexError: if position is out of bounds
+        """
+        return memview[pos]
+else:
+    def _read_byte(memview, pos):
+        """ Read a byte from memoryview as an integer
+
+            Raises:
+                IndexError: if position is out of bounds
+        """
+        return ord(memview[pos])
+
+
+def encode_varint(num):
```

Comment:
Are VARINTs used elsewhere in the new Kafka protocol APIs? Perhaps these functions belong under kafka.protocol.types ?

Reply:
No, currently only the Record structure uses it. If other API's will use it we can restructure later. I want this module to be very isolated with a good Facade. I want to cythonize this part later, I already did some work on that in `aiokafka`, gives very good results so far.